### PR TITLE
Fix handling of unlimited (-1) ulimit values

### PR DESCRIPTION
**Pull request by thaJeztah**
Addresses https://github.com/moby/moby/issues/39125

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @justincormack @AkihiroSuda @vdemeester PTAL

File: ulimit.go [hunk 1]
```diff
@@ -96,7 +96,11 @@ func ParseUlimit(val string) (*Ulimit, error) {
...
 	}
 
-	if soft > *hard {
+	if soft == -1 && *hard != -1 {
+		return nil, fmt.Errorf("ulimit soft limit must be less than or equal to hard limit: soft: -1 (unlimited), hard: %d", *hard)
+	}
+
+	if *hard != -1 && soft > *hard {
 		return nil, fmt.Errorf("ulimit soft limit must be less than or equal to hard limit: %d > %d", soft, *hard)
 	}
```
Comment (AkihiroSuda) [inline ulimit.go]: LGTM, but may be the code can be written as 
```go
if *hard != -1 {
  if soft == -1 { ... }
  if soft > *hard { ... }
}
```

Comment (thaJeztah) [reply ulimit.go]: Ah, was actually thinking about that after posting; I can update 👍

**Comment by thaJeztah** [comment]
@AkihiroSuda updated; ptal

**Pull request merged**